### PR TITLE
bam to fastq

### PR DESCRIPTION
**Pull request by rlancaster96**
- adds the bam to fastq functionality using samtools sort, collate, and fastq in `fastq_prep.nf`

Weird bug for me that might not be the same on other people's setups: samtools collate has a hard time finding the correct /tmp directory, so I have to specify it before submitting the nextflow job. The error associated with this is: 
```
Failed to open file "/home/groups/CEDAR/lancasru/tmp/collatef.0000.bam" : No such file or directory
```
and the 2 fastq files written will be empty.

resolved by slurm setting TMPDIR to /tmp

### CODE REVIEW

File: workflows/fastq_prep/fastq_prep.nf [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env nextflow
+
+// before running, make sure the tmp directory is set as /tmp
+// > cd <work directory>
+// > export TMPDIR=/tmp
```
Comment (elisabethgoldman) [inline workflows/fastq_prep/fastq_prep.nf:5]: I know `export TMPDIR=/tmp` is something we often set manually prior to running workflows with singularity involved; is it possible to pre-set this in the global nextflow.config file, e.g., 
```
process {
    beforeScript = 'TMPDIR=$(mktemp -d); export TMPDIR'
}
```
or, potentially as a per process option if necessary? 
```
process TOFASTQ {
    beforeScript = 'TMPDIR=$(mktemp -d); export TMPDIR'
```
If it's not a quick fix, can leave it manually for time being

**Review by elisabethgoldman — APPROVED** [review]
One small comment about automating setting TMPDIR, if it's an easy possibility. If it's not simple as it seems, completely fine to leave it be for now

**Comment by elisabethgoldman** [comment]
> * adds the bam to fastq functionality using samtools sort, collate, and fastq in `fastq_prep.nf`
> 
> Weird bug for me that might not be the same on other people's setups: samtools collate has a hard time finding the correct /tmp directory, so I have to specify it before submitting the nextflow job. The error associated with this is:
> 
> ```
> Failed to open file "/home/groups/CEDAR/lancasru/tmp/collatef.0000.bam" : No such file or directory
> ```
> 
> and the 2 fastq files written will be empty.
> 
> resolved by slurm setting TMPDIR to /tmp

I submitted my review and then re-read this comment -- totally fine if TMPDIR needs to be set manually; the tmp directory needing to be set (or else causing error) is not uncommon in Singularity and it sounds like its already been buggy

File: workflows/fastq_prep/fastq_prep.nf (hunk 1 above)
Comment (rlancaster96) [reply workflows/fastq_prep/fastq_prep.nf:5]: thank you! this is a great solution, I updated the TOFASTQ process to set TMPDIR=/tmp in the beforescript block, and it runs successfully regardless of what TMPDIR is set to for your bash global variable.

**Pull request merged**